### ISSUE TITLE
Creating post rules for large number of post codes

### ISSUE THREAD

**Issue by sangrambharat** [christianblais/Parcelify#33]
I have an extensive list of 16000 post codes. I need to create shipping rules for three shipping rates of $10, $20 and $40 for these postcodes. What type of rule should I create? Is it possible to use 'include' with such large number of postcodes? 
Thanks.

**Comment by christianblais** [comment]
Are they all unique, or can you not come up with some reusable patterns, such as "starts with 902", or "include 902[5-9][1-3]" (that would be 902, followed by 5, 6, 7, 8, or 9, then followed by 1, 2, or 3. Otherwise, 16000 is too much. That will put too much pressure on the free service I offer and will require you to run the software on your own machine.

**Comment by sangrambharat** [comment]
Thanks for the quick reply and suggestion.
I removed the duplicate entries and now I am left with 2900 postcodes.
[postcodes unique.xlsx](https://github.com/christianblais/Parcelify/files/1497514/postcodes.unique.xlsx)

For some of them I can apply the rule you suggested but there are many that are not following a pattern.
Please suggest.

**Comment by christianblais** [comment]
I see a lot of patterns in there. I cannot help you with the entirety of the list, but say we take the postal codes in the 2000 ranges, you want them all from 2000 up to 2899, with few exceptions, so you can accomplish that with two filters, like;

start with `2[0-9]{3}` which means 2 followed by any digits, 3 times, then you add another filter for exclude `2091|2851|285[3-9]|286[0-2]|289[2-7]` etc to exclude the few that aren't there. Again, I'm not here to teach you about regular expressions, but I believe it can be done.

Good luck!

**Comment by sangrambharat** [comment]
Thanks Christian. 
With "start with", I am using this value 0200|0822|0852|1340|1440|1481|1685|1701|1871|2[0-9]{3}|3[0-9]{3}|4[0-9]{3}|5[0-9]{3}
and it is working fine.

But as soon as I use another filter "exclude" with value 2003|2005|205[1|3|4|6|8]|2078|2091|2098|21[23|39|49|69]|218[0-9]|22[01|02|15]|22[35-49]|22[52-55]|2266|22[68-77]|2279|2288|23[01|09|10|11|13|28|29|31|32]23[36-39]|23[41-49]|23[51-59]|23[61-69]|23[71-79]|2381|23[83-89]|23[93-99]|24[01-25]|24[31-38]
sometimes it picks shipping rate correctly but sometimes it shows "shipping not defined" for the same postcode.

Is the exclusion list too long or is there any error in syntax?
I know I am giving you pain but this is very important for me. Thanks a lot.

**Comment by christianblais** [comment]
I don't think you can use things like [31-39]. It only works for single digits 0-9.
